core/src/main/java/org/apache/spark/shuffle/sort/ShuffleInMemorySorter.java

```diff
@@ -32,24 +37,39 @@ public int compare(PackedRecordPointer left, PackedRecordPointer right) {
   }
   private static final SortComparator SORT_COMPARATOR = new SortComparator();
 
+  private final MemoryConsumer consumer;
+  private final TaskMemoryManager memoryManager;
+
   /**
    * An array of record pointers and partition ids that have been encoded by
    * {@link PackedRecordPointer}. The sort operates on this array instead of directly manipulating
    * records.
    */
-  private long[] array;
+  private LongArray array;
 
   /**
    * The position in the pointer array where new records can be inserted.
    */
   private int pos = 0;
 
-  public ShuffleInMemorySorter(int initialSize) {
+  public ShuffleInMemorySorter(
+      MemoryConsumer consumer,
+      TaskMemoryManager memoryManager,
```

Comment:
Why does this class take both a `memoryManager` _and_ a consumer? Why not pass it just the consumer and use methods of the `consumer` to do the allocation?